bundles/org.openhab.binding.sbus/src/main/java/org/openhab/binding/sbus/handler/AbstractSbusHandler.java

```diff
@@ -0,0 +1,151 @@
+        if (bridgeHandler == null || bridgeHandler.getThing().getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Bridge is not online");
+            return;
+        }
+
+        try {
+            sbusAdapter = bridgeHandler.getSbusConnection();
+            updateStatus(ThingStatus.ONLINE);
+            startPolling();
+        } catch (Exception e) {
+            logger.error("Error initializing SBUS connection", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    /**
+     * Initialize channels for this device based on its configuration.
+     * This method should be implemented by concrete handlers to set up their specific channels.
+     */
+    protected abstract void initializeChannels();
+
+    /**
+     * Create or update a channel with the specified ID and type.
+     *
+     * @param channelId The ID of the channel to create/update
+     * @param channelTypeId The type ID of the channel
+     */
+    protected void createChannel(String channelId, String channelTypeId) {
+        ThingBuilder thingBuilder = ThingBuilder.create(getThing().getThingTypeUID(), getThing().getUID())
+                .withConfiguration(getThing().getConfiguration()).withBridge(getThing().getBridgeUID());
+
+        // Add all existing channels except the one we're creating/updating
+        ChannelUID newChannelUID = new ChannelUID(getThing().getUID(), channelId);
+        for (Channel existingChannel : getThing().getChannels()) {
+            if (!existingChannel.getUID().equals(newChannelUID)) {
+                thingBuilder.withChannel(existingChannel);
+            }
+        }
+
+        // Add the new channel
+        Channel channel = ChannelBuilder.create(newChannelUID).withType(new ChannelTypeUID(BINDING_ID, channelTypeId))
+                .withConfiguration(new Configuration()).build();
+        thingBuilder.withChannel(channel);
+
+        // Update the thing with the new channel configuration
+        updateThing(thingBuilder.build());
+    }
+
+    /**
+     * Start polling the device for updates based on the configured refresh interval.
+     */
+    protected void startPolling() {
+        SbusDeviceConfig config = getConfigAs(SbusDeviceConfig.class);
+        if (config.refresh > 0) {
+            pollingJob = scheduler.scheduleWithFixedDelay(() -> {
+                try {
+                    pollDevice();
+                } catch (Exception e) {
+                    logger.error("Error polling SBUS device", e);
+                }
+            }, 0, config.refresh, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Poll the device for updates. This method should be implemented by concrete handlers
+     * to update their specific channel states.
+     */
+    protected abstract void pollDevice();
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
```

Comment:
this throws a warning, can be fixed similar to the sbus Adpater below (by introducing a local var)